### PR TITLE
Fix overflow of line in page-break-lines-mode when using line numbers

### PR DESCRIPTION
**Pull request by Compro-Prasad**
This is picked up from

https://github.com/purcell/page-break-lines/blob/fd3b7e38ad8747cd009ead7ef1bb150849ccc693/page-break-lines.el

Do check the full diff as it contains many changes about versions(its 0?).

@bmag , @syl20bnr

### CODE REVIEW

**Comment by sdwolfz** [comment]
Wow, I always (wrongly) assumed the `^L` display as a line was done by emacs by default. I tried it out and it does fix the issue with line numbers, but looks like it still has problems with the git gutter. Anyway it's an external package so these issues should be fixed upstream. I'll merge in a few days if nobody else does it before I can.

**Comment by Compro-Prasad** [comment]
Its preferable to use the package itself. It may fix changes that may come up in future. This way we don't need to modify multiple times as the actual package changes.

**Comment by Compro-Prasad** [comment]
The CI is fixed. Wonderful. Looks premium now.

**Comment by syl20bnr** [comment]
The reason why it is in core is because we need it in the home buffer, so this is just aesthetic reason :-) I guess we can live with updating it once in a while. Nice PR 👍

File: core/libs/page-break-lines.el [hunk 1]
```diff
@@ -1,11 +1,11 @@
-;;; page-break-lines.el --- Display ugly ^L page breaks as tidy horizontal lines
```
Comment (syl20bnr) [inline core/libs/page-break-lines.el]: ahah, funny change :-)

**Comment by purcell** [comment]
> Its preferable to use the package itself. It may fix changes that may come up in future. This way we don't need to modify multiple times as the actual package changes.

Yeah, I agree - it's unclear why this package would need to be vendored here. The vendoring led to @Compro-Prasad filing an issue upstream for something that had already been resolved.

**Comment by sdwolfz** [comment]
I think it was vendored since the functionality was needed even before packages were installed.

**Comment by sdwolfz** [comment]
Thank you :heart:!
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**